routes/post.go

```diff
@@ -1599,17 +1600,601 @@ func (fes *APIServer) GetDiamondedPosts(ww http.ResponseWriter, req *http.Reques
+		SenderProfileEntryResponse:   _profileEntryToResponse(senderProfileEntry, fes.Params, verifiedMap, utxoView),
 	}
 	if err = json.NewEncoder(ww).Encode(res); err != nil {
 		_AddBadRequestError(ww, fmt.Sprintf("GetDiamondedPosts: Problem encoding response as JSON: %v", err))
 		return
 	}
 }
+
+type GetLikesForPostRequest struct {
+	// PostHashHex to fetch.
+	PostHashHex                string `safeForLogging:"true"`
+	Offset                     uint32 `safeForLogging:"true"`
+	Limit                      uint32 `safeForLogging:"true"`
+	ReaderPublicKeyBase58Check string `safeForLogging:"true"`
+}
+
+type GetLikesForPostResponse struct {
+	Likers []*ProfileEntryResponse
+}
+
+func (fes *APIServer) GetLikesForPost(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetLikesForPostRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("GetLikesForPost: Problem parsing request body: %v", err))
+		return
+	}
+
+	// Decode the postHash.
+	var postHash *lib.BlockHash
+	if requestData.PostHashHex == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Must provide a PostHashHex to fetch."))
+		return
+	} else {
+		postHashBytes, err := hex.DecodeString(requestData.PostHashHex)
+		if err != nil || len(postHashBytes) != lib.HashSizeBytes {
+			_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error parsing post hash %v: %v",
+				requestData.PostHashHex, err))
+			return
+		}
+		postHash = &lib.BlockHash{}
+		copy(postHash[:], postHashBytes)
+	}
+
+	// Decode the reader public key into bytes. Default to nil if no pub key is passed in.
+	var readerPublicKeyBytes []byte
+	if requestData.ReaderPublicKeyBase58Check != "" {
+		var err error
+		readerPublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.ReaderPublicKeyBase58Check)
+		if requestData.ReaderPublicKeyBase58Check != "" && err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Problem decoding user public key: %v : %s", err,
+				requestData.ReaderPublicKeyBase58Check))
+			return
+		}
+	}
+
+	// Get a view with all the mempool transactions.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error constucting utxoView: %v", err))
+		return
+	}
+
+	// Fetch the likers for the post requested.
+	reclouterPubKeys, err := utxoView.GetLikesForPostHash(postHash)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetLikesForPost: Error getting likers %v", err))
+		return
+	}
+
+	// Filter out any restricted profiles.
+	addReaderPublicKey := false
+	pkMapToFilter := make(map[lib.PkMapKey][]byte)
+	for _, pubKey := range reclouterPubKeys {
+		pkMapKey := lib.MakePkMapKey(pubKey)
+		pkMapToFilter[pkMapKey] = pubKey
+		if reflect.DeepEqual(pubKey, readerPublicKeyBytes) {
```

Comment:
can we skip this reflect.DeepEqual call if we have `addReaderPublicKey = true` already? `reflect.DeepEqual` is relatively expensive so if we can avoid that call, it would help performance. 

Actually it would probably be best if we look up the like entry directly for the reader instead of comparing every public key to the reader.
```
	postEntry := utxoView.GetPostEntryForPostHash(postHash)
	postEntryReaderState := utxoView.GetPostEntryReaderState(readerPublicKeyBytes, postEntry)
	addReaderPublicKey := postEntryReaderState.LikedByReader
```